plugin/lighthouse/handler_test.go

```diff
@@ -160,90 +157,72 @@ func testNext() {
 		rec = dnstest.NewRecorder(&test.ResponseWriter{})
 	})
 
-	When("Zones not configured", func() {
-		BeforeEach(func() {
-			tc = test.Case{
-				// service does not exist, expect error.
-				Qname: "mysvc2.default.svc.cluster.local.", Qtype: dns.TypeA,
+	When("type A DNS query for a non-matching lighthouse zone and matching fallthrough zone", func() {
+		It("should invoke the next plugin", func() {
+			lh.Fall = fall.F{Zones: []string{"cluster.local.", "cluster.east."}}
+			executeTestCase(lh, rec, test.Case{
+				Qname: service1 + "." + namespace1 + ".svc.cluster.east.",
+				Qtype: dns.TypeA,
 				Rcode: dns.RcodeBadCookie,
-			}
+			})
 		})
-		It("Should call next plugin", func() {
-			m := tc.Msg()
-			code, err := lh.ServeDNS(context.TODO(), rec, m)
-			Expect(err).To(HaveOccurred())
-			Expect(code).Should(Equal(tc.Rcode))
+	})
+
+	When("type A DNS query for a non-matching lighthouse zone and non-matching fallthrough zone", func() {
```

Comment:
Do we have a test case when no explicit fallthrough zone is configured? It means "all zones match" and it should fallthrough. I think we had test case for it, just wanted to confirm.

Reply:
So I added a new test case with `lh.Fall = fall.F{}` but it doesn't fall through. Looking at the `Fall.Through` code, if `Zones` is empty then `Matches` returns `""` which fails the check `!= ""`. So there must be at least one matching zone.  Is this what we expect?